### PR TITLE
[#566] Jenkins build pipeline for kubernetes infrastructure

### PR DESCRIPTION
**Pull request by serglupach**
#566 - 1st item: build of 3d party customised docker containers.

### CODE REVIEW

File: deploy/build-dex.Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    options{
+        buildDiscarder(logRotator(numToKeepStr: '5'))
+        disableConcurrentBuilds()
+    }
+    parameters {
+        booleanParam(defaultValue: false, description: 'Enable slack notifications', name: 'EnableSlackNotifications')
+        booleanParam(defaultValue: false, description: '', name: 'EnableDockerCache')
+    }
+    environment {
```
Comment (alexander-semenets) [inline deploy/build-dex.Jenkinsfile]: Currently all such properties are stored in job parameters section and not commited to git within pipeline.
So we don't share our endpoint or resources names or anything else. 
Shouldn't we follow some common way and set this properties in jenkins job as parameters the same way?
@kirillmakhonin @dsuslov

File: k8s/k8s-infra/oauth2-proxy-docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+RUN   git config --global user.email "you@example.com" \
+&& git config --global user.name "Your Name"
+
+RUN go get -d github.com/bitly/oauth2_proxy
+RUN cd /go/src/github.com/bitly/oauth2_proxy \
+&& git remote add JoelSpeed https://github.com/JoelSpeed/oauth2_proxy.git \
+&& git pull -r JoelSpeed kubernetes \
+&& go install
+
+FROM alpine:latest
```
Comment (alexander-semenets) [inline k8s/k8s-infra/oauth2-proxy-docker/Dockerfile]: I guess we should pin some version here

File: deploy/build-dex.Jenkinsfile (hunk 1 above)
Comment (serglupach) [reply deploy/build-dex.Jenkinsfile]: Has Removed parameters in pipeline,  configured them in job parameters section.

File: deploy/build-dex.Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                    '''
+                }
+            }
+        }
+        stage('Push docker image Dex') {
+            steps {
+                sh """
+                docker tag ${docker_registry}/${dex_dockerimage}:${BUILD_NUMBER} ${docker_registry}/${dex_dockerimage}:latest
+                docker tag ${docker_registry}/${dex_dockerimage}:${BUILD_NUMBER} ${docker_registry}/${dex_dockerimage}:${BUILD_NUMBER}-${commit_id}
+                docker push ${docker_registry}/${dex_dockerimage}:${BUILD_NUMBER}-${commit_id}
```
[the remark this answers is not in the document]
Comment (serglupach) [reply deploy/build-dex.Jenkinsfile]: Updated pipeline. Added script for legion-compatible version, and updated all steps.

File: k8s/k8s-infra/oauth2-proxy-docker/Dockerfile (hunk 2 above)
Comment (serglupach) [reply k8s/k8s-infra/oauth2-proxy-docker/Dockerfile]: Configured latest version of base dockerimage alpine:3.8

File: deploy/build-k8s-infra.Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+                }
+            }
+        }
+        stage('Build docker images') {
+            parallel {
+                stage('Build kube-fluentd') {
+                    steps {
+                        dir ("${build_workspace}/kube-fluentd") {
+                            sh """
+                            pwd
```
[the remark this answers is not in the document]
Comment (serglupach) [reply deploy/build-k8s-infra.Jenkinsfile]: Cleaned code from debug elements

File: deploy/build-k8s-infra.Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+                            ls -alh
+                            docker build ${Globals.dockerCacheArg} -t legion/${kube_fluentd_dockerimage}:${Globals.buildVersion} ${Globals.dockerLabels} -f Dockerfile .
+                            """
+                        }
+                    }
+                }
+                stage('Build kube-elb-security') {
+                    steps {
+                        dir ("${build_workspace}/kube-elb-security") {
+                            sh """
```
[the remark this answers is not in the document]
Comment (serglupach) [reply deploy/build-k8s-infra.Jenkinsfile]: Cleaned code from debug elements

**Pull request closed without merging**